agent/eni/statemanager/statemanager_linux.go

```diff
@@ -16,159 +16,85 @@
 package statemanager
 
 import (
-	"sync"
-
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
 	log "github.com/cihub/seelog"
 	"github.com/vishvananda/netlink"
 )
 
-const mapCapacityHint = 10 // mapCapacityHint is a nominal hint for map capacity
-
-// stateChangeUnsafe exposes methods that are unsafe to use
-// without grabbing locks in the correct order
-type stateChangeUnsafe interface {
-	AddDeviceWithMACAddressUnsafe(deviceName, macAddress string)
-	RemoveDeviceWithMACAddressUnsafe(mac string)
-	RemoveDeviceUnsafe(deviceName string)
-}
-
-// StateManager interface incorporates stateChangeUnsafe and adds
-// additional methods to lock, unlock and retrieve all state
 type StateManager interface {
-	stateChangeUnsafe
-	GetAll() map[string]string
 	Init(state []netlink.Link)
 	Reconcile(currentState map[string]string)
-	AddDeviceWithMACAddress(deviceName, macAddress string)
-	RemoveDeviceWithMACAddress(mac string)
-	RemoveDevice(deviceName string)
-	IsMACAddressPresent(macAddress string) bool
+	HandleENIEvent(mac string)
 }
 
-// stateManager holds the state of ENIs connected to the instance
+// stateManager handles the state change of eni
 type stateManager struct {
-	updateLock sync.RWMutex
-	enis       map[string]string // enis is a map of MAC => Device-Name
+	agentState dockerstate.TaskEngineState
+	taskEvent  chan api.TaskStateChange
 }
 
 // New returns a new StateManager
-func New() StateManager {
+func New(state dockerstate.TaskEngineState, event chan api.TaskStateChange) StateManager {
```

Comment:
None of the methods in this file are covered by unit tests. Please add test coverage for these methods.